Objective-C/CBLReplicator.mm

```diff
@@ -494,14 +510,20 @@ - (BOOL) isDocumentPending: (NSString*)documentID error: (NSError**)error {
         return false;
     }
     
-    if (!_repl) {
-        CBLLogInfo(Sync, @"Trying to fetch document pending status without a c4replicator %@", _repl);
+    C4Error c4err = {};
+    C4Replicator* repl = [self c4repl: &c4err];
+    if (!repl) {
+        NSError* err = nil;
+        convertError(c4err, &err);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %@", self, err.localizedDescription);
+        if (error)
+            *error = err;
         return false;
     }
     
-    C4Error c4err = {};
+    c4err = {};
     CBLStringBytes docID(documentID);
-    BOOL isPending = c4repl_isDocumentPending(_repl, docID, &c4err);
+    BOOL isPending = c4repl_isDocumentPending(repl, docID, &c4err);
```

Comment:
Should we call this in CBL_LOCK(self) to avoid race condition if repl is freed when stopped or offline?